code/modules/atmospherics/machinery/portable/canister.dm

```diff
@@ -73,6 +79,12 @@
 	AddElement(/datum/element/atmos_sensitive, mapload)
 	AddElement(/datum/element/volatile_gas_storage)
 	AddComponent(/datum/component/gas_leaker, leak_rate=0.01)
+	register_context()
+
+/obj/machinery/portable_atmospherics/canister/Destroy()
+	QDEL_NULL(wires)
+	QDEL_NULL(rig)
```

Comment:
The reference to the rig will be then cleared as it's moved out of the canister (In general, all contents of a movable are _moved_ to nullspace and then deleted when their holder is deleted).